src/erldns_encoder.erl

```diff
@@ -27,7 +27,17 @@
 %% exception.
 -spec encode_message(dns:message()) -> dns:message_bin().
 encode_message(Response) ->
-  encode_message(Response, []).
+  case application:get_env(erldns, catch_exceptions) of
```

Comment:
This appears to be inlining the code from `encode_message/2`. I'm uncertain about the reason for replacing the call to it.
Is there a big difference between 
`dns:encode_message/1` and `dns:encode_message/2` with Opts = []?

Reply:
There *is* a big difference between `dns:encode_message/1` and `dns:encode_message/2`, the later performing truncation based on the truncation rules provided.

Reply:
![image](https://user-images.githubusercontent.com/3962/41918676-6e8c23ac-7922-11e8-93be-9463e9b6fbe5.png)

Reply:
I didn't fully appreciate the distinction into now either. 😉

Reply:
I missed the opportunity to truncate that image